### PR TITLE
mobile: skipping command line parsing for E-M start up

### PR DESCRIPTION
**Pull request by alyssawilk**
This is a slightly cleaner code path setting options directly, rather than turning into command line options and back, and may provide a minor start up boost.

Risk Level: n/a for Envoy, medium for E-M
Testing: options test verifies options remain consistent 
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by alyssawilk** [comment]
Ok unit tests fixed up - PTAL!

File: source/server/options_impl.h [hunk 1]
```diff
@@ -54,6 +54,9 @@ class OptionsImpl : public Server::Options, protected Logger::Loggable<Logger::I
   OptionsImpl(const std::string& service_cluster, const std::string& service_node,
               const std::string& service_zone, spdlog::level::level_enum log_level);
 
+  // Constructor for mobile
+  OptionsImpl() {}
```
Comment (RyanTheOptimist) [inline source/server/options_impl.h:58]: nit: = default is, I think, preferred though I honestly don't remember why.

**Review by RyanTheOptimist — DISMISSED** [review]
Wow, this is great! I'm surprised that it's this simple.

**Review by RyanTheOptimist — DISMISSED** [review]

**Comment by jpsim** [comment]
Could this assertion error in the Swift tests have anything to do with this change?

> [external/envoy/source/extensions/listener_managers/listener_manager/listener_impl.cc:119] assert failure: sockets_.size() == num_sockets.

**Review by jpsim — APPROVED** [review]
👍 if we can confirm that the Swift tests assertion failure is unrelated.

**Comment by jpsim** [comment]
Hmmm, the kotlin tests have some concerning failures too: https://github.com/envoyproxy/envoy/actions/runs/3959781241/jobs/6783722362

**Review by RyanTheOptimist — APPROVED** [review]

**Comment by alyssawilk** [comment]
Thanks for the call-out on swift/kotlin @jpsim it was an indeed a bug fixed and regression-tested with ConcurrencyZeroIsOne
This is now passing all E-M builds so I'm going to enable auto-merge.

**Review by jpsim — APPROVED** [review]

**Pull request merged**